packages/ui/src/components/multi-select-v2.tsx

```diff
@@ -0,0 +1,329 @@
+                }
+              }
+              const newOptions = [...(isControlled ? value : selected), newOption]
+              if (isControlled) {
+                onChange?.(newOptions)
+              } else {
+                setSelected(newOptions)
+              }
+              setInputValue('')
+              setSearchQuery?.('')
+              e.preventDefault()
+            }
+          }
+          if (e.key === 'Escape') {
+            input.blur()
+          }
+        }
+      },
+      [handleUnselect, selected, disallowCreation, onChange, options, setInputValue, isControlled, value]
+    )
+
+    useEffect(() => {
+      if (open) {
+        document.addEventListener('mousedown', handleClickOutside)
+        document.addEventListener('touchend', handleClickOutside)
+      } else {
+        document.removeEventListener('mousedown', handleClickOutside)
+        document.removeEventListener('touchend', handleClickOutside)
+      }
+
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside)
+        document.removeEventListener('touchend', handleClickOutside)
+      }
+    }, [open])
+
+    useEffect(() => {
+      setOptions(arrayOptions ?? [])
+    }, [arrayOptions])
+
+    const [availableOptions, setAvailableOptions] = useState<MultiSelectOption[]>([])
+
+    useEffect(() => {
+      if (!options || options.length === 0) {
+        setAvailableOptions([])
+        return
+      }
+
+      const filteredOptions = options.filter(option =>
+        isControlled
+          ? !value?.some(selectedOption => selectedOption.id === option.id)
+          : !selected?.some(selectedOption => selectedOption.id === option.id)
+      )
+
+      setAvailableOptions(filteredOptions)
+    }, [options, selected, isControlled, value, inputValue, searchQuery, open])
+    return (
+      <div className="cn-multi-select-outer-container">
+        <Label disabled={disabled}>{label}</Label>
+        <Command.Root
+          ref={dropdownRef}
+          {...commandProps}
+          onKeyDown={e => {
+            handleKeyDown(e)
+            commandProps?.onKeyDown?.(e)
+          }}
+          // filter={(_value, _search) => {
+          //   return 1
+          // }}
+          shouldFilter={false}
+          className={cn('h-auto overflow-visible bg-transparent', commandProps?.className)}
+        >
+          <div
+            className={cn('cn-multi-select-container', className)}
+            onClick={() => {
+              if (disabled) return
+              inputRef?.current?.focus()
+            }}
+            onKeyDown={noop}
+            role="textbox"
+            tabIndex={disabled ? -1 : 0}
+            aria-label={placeholder}
+          >
+            <div className="cn-multi-select-tag-wrapper">
+              {(isControlled ? value : selected).map(option => {
+                return (
+                  <Tag
+                    id={String(option.id)}
+                    key={option.id}
```

Comment:
not blocker: but let's check if we can avoid id. maybe use `key+value`? might need to discuss with BE/product for API confirmation